rtengine/iplocallab.cc

```diff
@@ -14276,13 +14278,50 @@ void ImProcFunctions::Lab_Local(
         return;
     }
 
-    //BENCHFUN
+
 
     constexpr int del = 3; // to avoid crash with [loy - begy] and [lox - begx] and bfh bfw  // with gtk2 [loy - begy-1] [lox - begx -1 ] and del = 1
     struct local_params lp;
     calcLocalParams(sp, oW, oH, params->locallab, lp, prevDeltaE, llColorMask, llColorMaskinv, llExpMask, llExpMaskinv, llSHMask, llSHMaskinv, llvibMask, lllcMask, llsharMask, llcbMask, llretiMask, llsoftMask, lltmMask, llblMask, lllogMask, ll_Mask, llcieMask, locwavCurveden, locwavdenutili);
 
     //avoidcolshi(lp, sp, transformed, reserved,  cy, cx, sk);
+    //BENCHFUN
+
+
+
+
+    //avoid negative values RGB then Lab when using before SE CBDL or Dehaze, or processor type...
+
+    int bw0 = transformed->W;
+    int bh0 = transformed->H;
+
+    float epsi0 = 0.000001f;
+    bool nocrash = false;
+    bool cbdl = false;
+    if(params->dirpyrequalizer.cbdlMethod == "bef") {//If user choose "after black and white" this function which removes negative values is not used, hence CBDL is best performed, after Selective Editing in Lab mode
+        cbdl = true;
+    }
+    bool dirpyrdeha = (params->dirpyrequalizer.enabled && cbdl)  ||  params->dehaze.enabled  || lp.avoidneg;//lp.avoidneg in setting 
```

Comment:
`dirpyrdeha` is just a duplicate of `nocrash`.

Reply:
ok